pkg/util/helper/binding.go

```diff
@@ -399,6 +399,10 @@ func GenerateReplicaRequirements(podTemplate *corev1.PodTemplateSpec) *workv1alp
 		return &workv1alpha2.ReplicaRequirements{
 			NodeClaim:       nodeClaim,
 			ResourceRequest: resourceRequest,
+			Namespace:       podTemplate.Namespace,
```

Comment:
Shall we add a feature gate check here? The two fields are unnecessary for users that do not enable this feature?

Reply:
By the way, we are sure that we have the opportunity to set the `PriorityClassName` in the `if nodeClaim != nil || resourceRequest != nil`,  because the `resourceRequest` can not be nil, right?

Reply:
Hi, I am not quite sure if I understand this question correct, I think we handle the nil replicaRequirement in estimator [here](https://github.com/karmada-io/karmada/blob/master/pkg/estimator/client/accurate.go#L88-L100)

The default `pb.ReplicaRequirements` is empty, it will be populated only if ReplicaRequirements is not nil
```
	req := &pb.MaxAvailableReplicasRequest{
		Cluster:             cluster,
		ReplicaRequirements: pb.ReplicaRequirements{},
	}
	if replicaRequirements != nil {
		req.ReplicaRequirements.ResourceRequest = replicaRequirements.ResourceRequest
		if replicaRequirements.NodeClaim != nil {
			req.ReplicaRequirements.NodeClaim = &pb.NodeClaim{
				NodeAffinity: replicaRequirements.NodeClaim.HardNodeAffinity,
				NodeSelector: replicaRequirements.NodeClaim.NodeSelector,
				Tolerations:  replicaRequirements.NodeClaim.Tolerations,
			}
		}
	}
```
If the pb.ReplicaRequirements is empty, which means empty ResourceList, the current [logic](https://github.com/karmada-io/karmada/blob/master/pkg/estimator/server/estimate.go#L98-L106) still calculate the replica
```
func (es *AccurateSchedulerEstimatorServer) nodeMaxAvailableReplica(node *framework.NodeInfo, rl corev1.ResourceList) int32 {
	rest := node.Allocatable.Clone().SubResource(node.Requested)
	// The number of pods in a node is a kind of resource in node allocatable resources.
	// However, total requested resources of all pods on this node, i.e. `node.Requested`,
	// do not contain pod resources. So after subtraction, we should cope with allowed pod
	// number manually which is the upper bound of this node available replicas.
	rest.AllowedPodNumber = util.MaxInt64(rest.AllowedPodNumber-int64(len(node.Pods)), 0)
	return int32(rest.MaxDivided(rl))
}
```